src/python/pants/core/target_types.py

```diff
@@ -155,28 +224,48 @@ def globs(self) -> tuple[str, ...]:
     @property
     def file_path(self) -> str:
         assert self.value
-        filename = self.value if isinstance(self.value, str) else self.value.filename
+        # NB: For http_source(s) we end with "/" and handle filename in the hydration rule
+        filename = self.value if isinstance(self.value, str) else ""
         return os.path.join(self.address.spec_path, filename)
 
 
 @rule_helper
-async def _hydrate_asset_source(request: GenerateSourcesRequest) -> GeneratedSources:
+async def _hydrate_asset_source(
+    request: GenerateSourcesRequest, platform: Platform
+) -> GeneratedSources:
     target = request.protocol_target
     source_field = target[AssetSourceField]
     if isinstance(source_field.value, str):
         return GeneratedSources(request.protocol_sources)
 
-    http_source = source_field.value
+    source = source_field.value
+    if isinstance(source, HTTPSources):
+        http_source = _HTTPInfo(**dataclasses.asdict(source)[platform.value])
```

Comment:
Why the roundtrip to `dict` land? The `HTTPSources.value` already is decoded into a `_HTTPInfo`, so couldn't this just be
```python
  http_source = getattr(source, platform.value)
```

Reply:
d'oh!